pyro/infer/autoguide/guides.py

```diff
@@ -347,6 +347,145 @@ def median(self, *args, **kwargs):
         return self(*args, **kwargs)
 
 
+class AutoNormal(AutoGuide):
+    """This is AutoNormal, but it uses the same shape constraint logic found
+    in AutoMultivariateNormal and AutoDiagonalNormal.
+    """
+    def __init__(self, model, init_loc_fn=init_to_median, prefix='auto'):
+        #  Does this accept a 'prefix' arg?  It doesn't look like it.
+        self.init_loc_fn = init_loc_fn
```

Comment:
It looks like you're not yet using `init_loc_fn`. I would recommend adding
```py
if init_loc_fn is not init_to_feasible:
    raise NotImplementedError("TODO")
```
and we can work on that in a follow-up PR (I'm happy to add support if you want).